projects/fw_103/src/main.c

```diff
@@ -0,0 +1,59 @@
+#include <stdint.h>  // for integer types
+
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"  // interrupts are required for soft timers
+#include "log.h"        // for printing
+#include "wait.h"       // for wait function
+
+GpioAddress adc_addr = {
+  .port = GPIO_PORT_A,
+  .pin = 6,
+};
+
+static void prv_read_adc(const GpioAddress *btn_addr_ptr, void *context) {
+  uint16_t adc_data = 0;
+  if (adc_read_converted_pin(adc_addr, &adc_data) == STATUS_CODE_OK) {
+    LOG_DEBUG("ADC data: %d\n", adc_data);
+  } else {
+    LOG_DEBUG("Failed to read adc data");
+  }
+}
+
+int main(void) {
+  gpio_init();
+  gpio_it_init();
+  interrupt_init();
```

Comment:
I think interrupt needs to be initialized before gpio interrupt

Reply:
Right, thanks